### DISCUSSION TITLE
pulsar flusher 时间问题

### DISCUSSION THREAD

**Discussion by wkai886** [alibaba/ilogtail#686] [Help] [answered]
基于最新的 main 代码 编译，commit ： d0d2be280d638c2f475fe778b5c888dee6ed7082 
基础镜像使用 Asia/ShangHai 时区，用 pulsar 的 fluster 去生产数据，消费发现 contents里面的 _time_ 时区不对。

消费可得数据:
{
    "contents":{
        "_source_":"stderr",
        "_time_":"2023-02-28T02:30:25.29898325Z",
        "app":"log-test",
        "content":"2023/02/28 10:30:25 INFO 当前时间戳： 1677551425",
        "group":"sssss",
        "system":"xxxxxx"
    },
    "tags":{
        "container.image.name":"harbor/business/log-test:e3a50af-220407-0157",
        "container.ip":"10.0.0.0",
        "container.name":"business",
        "host.ip":"10.0.0.0",
        "hostname":"op-xxxxxx",
        "k8s.namespace.name":"ttttttt",
        "k8s.node.ip":"10.0.0.0",
        "k8s.node.name":"op-xxxx",
        "k8s.pod.name":"log-test.sssss-n4whw",
        "k8s.pod.uid":"1ac1a191-b4c8-4235-996d-a449e6880c4sss9"
    },
    "time":1677551425
}

contents 里面的 time 的时区不对， contents外面的 time 的时间是对的

**Comment by henryzhx8** [comment]
你是说"time":"2023-02-28T02:30:25.29898325Z"这个时间戳不对？这个时间戳和"time":1677551425不是代表同一个时间吗？

**Comment by shalousun** [comment, marked as answer]
2023-02-28T02:30:25.29898325Z中T就代表着是UTC时间，而content里你日志的时间是中国上海时间，utc时间+8就是中国上海时间。所以这两个是一样的
